AdminUi/src/AdminUi/ClientApp/src/app/components/client/client-list/client-list.component.ts

```diff
@@ -74,17 +79,67 @@ export class ClientListComponent {
             });
     }
 
-    pageChangeEvent(event: PageEvent) {
+    pageChangeEvent(event: PageEvent): void {
         this.pageIndex = event.pageIndex;
         this.pageSize = event.pageSize;
         this.getPagedData();
     }
 
-    dateConvert(date: any) {
+    dateConvert(date: any): string {
         return new Date(date).toLocaleDateString();
     }
 
-    addClient() {
+    addClient(): void {
         this.router.navigate([`/clients/create`]);
     }
+
+    deleteClient(): void {
+        this.loading = true;
+        let observableBatch: Observable<any>[] = [];
+        this.selection.selected.forEach(item => {
+            observableBatch.push(this.clientService.deleteClient(item.clientId));
+        });
```

Comment:
Not quite sure if I understood.
Would you rather I change the backbone to receive a list of IDs and use a single request?
Currently, we have a request per delete operation.
I added a multiselect behavior to the page so that, when you select multiple clients and delete, the requests are sent in parallel.

Reply:
> I added a multiselect behavior to the page so that, when you select multiple clients and delete, the requests are sent in parallel.
Okay, that explains it. I didn't see the stupid forEach method call - which is why I don't like.forEach ;-)